src/libpsl-native/src/getinodedata.cpp

```diff
@@ -0,0 +1,56 @@
+//! @file getinodedata.cpp
+//! @author Jeff Bienstadt <v-jebien@microsoft.com>
+//! @brief Retrieve the device ID and inode number of a file
+
+#include "getinodedata.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <locale.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <string>
+
+//! @brief GetInodeData retrieves a file's device and inode information.
```

Comment:
This one describes the function. The one at the top of the file describes the file.

I used the existing `getlinkcount.cpp` as the pattern for this.